src/combinatorics/configuration.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import combinatorics.hall.basic
+import data.fintype.card
+
+/-!
+# Configurations of Points and lines
+This file introduces abstract configurations of points and lines, and proves some basic properties.
+
+## Main definitions
+* `configuration`: Finite collections of points and lines with an incidence relation.
+* `configuration.dual`: The dual configuration obtained by swapping points and lines.
+* `configuration.nondegenerate`: Excludes certain degenerate configurations,
+  and imposes uniqueness of intersection points.
+* `configuration.has_points`: A nondegenerate configuration in which
+  every pair of lines has an intersection point.
+* `configuration.has_lines`:  nondegenerate configuration in which
+  every pair of points has a line through them.
+
+## Todo
+* Abstract projective planes.
+-/
+
+open_locale big_operators
+
+universe u
+
+/-- A configuration is a finite collections of points and lines with an incidence relation. -/
+structure configuration :=
+(P : Type u)
+(fP : fintype P)
+(L : Type u)
+(fL : fintype L)
+(R : P → L → Prop)
```

Comment:
The unbundled version I'm asking for a comparison with in my top-level comment is:
```lean
structure configuration (P : Type u) [fintype P] (L : Type u) [fintype L] :=
(R : P → L → Prop)
```
or maybe even just
```lean
def configuration (P : Type u) [fintype P] (L : Type u) [fintype L] := P → L → Prop
```